src/main/java/gg/uhc/uhc/modules/health/HealthCommand.java

```diff
@@ -48,39 +52,133 @@
                 .withValuesConvertedBy(new OnlinePlayerConverter());
 
         nonOptionsTabComplete = new NonDuplicateTabComplete(OnlinePlayerTabComplete.INSTANCE);
+
+        allPlayersSpec = parser
+                .acceptsAll(ImmutableList.of("a", "all"), "Check/modify all online players");
+        silentSpec = parser
+                .acceptsAll(ImmutableList.of("s", "silent"), "Do not notify players about their health being changed");
+
+        setCurrentHealthSpec = parser
+                .acceptsAll(ImmutableList.of("c", "current"), "Set the health of players")
+                .withRequiredArg()
+                .withValuesConvertedBy(new DoubleConverter()
+                        .setPredicate(DoublePredicates.GREATER_THAN_ZERO)
+                        .setType("New player health")
+                );
+
+        setMaximumHealthSpec = parser
+                .acceptsAll(ImmutableList.of("m", "maximum"), "Set the maximum health of players")
+                .withRequiredArg()
+                .withValuesConvertedBy(new DoubleConverter()
+                        .setPredicate(DoublePredicates.GREATER_THAN_ZERO)
+                        .setType("New maximum player health")
+                );
     }
 
     @Override
     protected boolean runCommand(CommandSender commandSender, OptionSet optionSet) {
-        List<Player> toShow = playerSpec.values(optionSet);
+        List<Player> players = playerSpec.values(optionSet);
+
+        if (optionSet.has(allPlayersSpec)) {
+            players = Lists.newArrayList(Bukkit.getOnlinePlayers());
+        }
 
-        if (toShow.size() == 0) {
+        if (players.size() == 0) {
             if (commandSender instanceof Player) {
-                toShow = Lists.newArrayList((Player) commandSender);
+                players = Lists.newArrayList((Player) commandSender);
             } else {
                 commandSender.sendMessage(messages.getRaw("no players"));
                 return true;
             }
         }
 
-        String playerPart = Joiner.on(messages.getRaw("player separator")).join(Iterables.transform(toShow, convert));
+        boolean isSettingCurrentHealth = optionSet.has(setCurrentHealthSpec);
+        boolean isSettingMaximumHealth = optionSet.has(setMaximumHealthSpec);
+        boolean isModification = isSettingCurrentHealth || isSettingMaximumHealth;
+
+        if (isModification && !commandSender.hasPermission(MODIFY_PERMISSION)) {
+            commandSender.sendMessage(messages.getRaw("cannot modify"));
+            return true;
+        }
+
+        // Set maximum health before current health in case of multiple operation
+        if (isSettingMaximumHealth) {
+            double newMaximumHealth = setMaximumHealthSpec.value(optionSet);
+
+            boolean limitedByServer = false;
+
+            for (Player player : players) {
+                player.setMaxHealth(newMaximumHealth);
+                double actualMaxHealth = player.getMaxHealth();
```

Comment:
Not really, if the maximum health is limited by the server setting  and checking if it actually got set to the expected value is probably the best way. Hooking into Spigot configuration itself sounds like a very nasty hack.